doc/sphinx/language/core/records.rst

```diff
@@ -177,6 +177,37 @@ Defining record types
          Class MyClass := { myfield2 : nat }.
          About myfield2. (* Argument name defaults to the class name and is marked implicit *)
 
+.. _record_where_clause:
+
+   .. example:: Using a :g:`where` clause in a record field
+
+      Records themselves do not support a :g:`where` clause, but record fields do,
+      with a somewhat unintuitive behaviour: the notation defined in the where clause
+      is active after the field where it appears, not in the field itself.
+
+      .. rocqtop:: all
+
+         Reserved Notation "a & b" (at level 40).
+         Record foo (op : nat -> nat -> nat) := mkfoo
+         {
+           point: nat where "a & b" := (op a b);
+           cond: forall a b, a & b = b & a
+         }.
+
+      However, it is not possible to add a :g:`where` clause for the record itself:
+
+      .. rocqtop:: all
+
+         Reserved Notation "a & b" (at level 40).
+         Fail Record foo (op : nat -> nat -> nat) := mkfoo
+         {
+           point: nat;
+           cond: forall a b, a & b = b & a
+         } where "a & b" := (op a b).
+
+   .. exn:: Error: "where" clause not supported for records.
+      :undocumented:
+
```

Comment:
Including examples of syntax that the documented grammar clearly doesn't support seems overly verbose.  I would drop this part (keep the `.. exn` part if you like).  There's an infinity of inputs that the grammar doesn't allow--why is this one so special?

Reply:
the grammar does allow it, which is why it produces a regular error and not a syntax error

Reply:
Not much point in describing syntax that always gets an error.  The parser parses it, but the edited grammar shown in this section does not (intentionally).  For which of these commands are decl_notations allowed?  Looks like only for `Inductive` and `CoInductive`.

```

inductive_or_record_definition: [
| opt_coercion cumul_ident_decl binders OPT [ "|" binders ] OPT [ ":" lconstr ] opt_constructors_or_fields decl_notations
]

gallina: [
| finite_token inductive_or_record_definition
| inductive_token LIST1 inductive_or_record_definition SEP "with"
]

finite_token: [
| "Variant"
| "Record"
| "Structure"
| "Class"
]

inductive_token: [
| "Inductive"
| "CoInductive"
]
```

Reply:
It's probably the first question the user will ask. I think it's better to document it.
And, the error was not documented before. It's better if it comes with an example.

Reply:
What's the question the user will ask?  The example doesn't address why it's an error.  The error message alone seems clear and it's also very clear from the syntax given in the document.  The example doesn't add anything to it.  It's very easy to describe the limitation in the command description.  Adding very obvious examples is not a service to the reader, particularly when they only show how to generate error messages--it's just more verbiage to wade through.

Reply:
I removed the failing example.